src/WinRT.Runtime/ExceptionHelpers.cs

```diff
@@ -403,6 +433,20 @@ public static unsafe void SetErrorInfo(Exception ex)
                             global::ABI.WinRT.Interop.ILanguageExceptionErrorInfo2Methods.CapturePropagationContext(languageErrorInfo2Ref, ex);
                         }
                     }
+                    else if (isLanguageException)
+                    {
+                        if (ex.Data.Contains("__RestrictedLanguageExceptionObjectReference") &&
+                            ex.Data["__RestrictedLanguageExceptionObjectReference"] is ObjectReference<IUnknownVftbl> objRef)
+                        {
+                            // Add ref as we are going to pass the responsibility for keeping the exception alive
+                            // to the error info.
+                            Marshal.AddRef(objRef.ThisPtr);

```

Comment:
This isn't clear to me, why do we need to addref here? Wouldn't the error info have already done an addref on the same pointer when we originally passed it to it? Am I missing some context?

Reply:
- The way this works is we first originate an error with the language exception.  At this point, the error info has a ref on the CCW of the exception.
- The error then propagates to the native side.
- Eventually lets say it comes back to C#.  We retrieve the error info and get the exception from it.  We then throw it which can go across threads and so on.  At the same time, if this exception were to go back to native code, we want to be able to set the same error info again so we store a reference to it in the exception data.  Now we got a cycle.  The error info keeps the exception CCW ref counted and the exception keeps the error info alive.
- To fix this, we release the ref held on the CCW of the exception by the error info.  Given the exception is now alive in C# code and is being thrown, that keeps it alive.
- Now when we come back to the ABI and need to set the error info again, we add ref again to make sure the error info keeps the exception alive.  At the same time, we remove our stored error info to remove the cycle.

Reply:
Removed the cycle handling logic as there is the scenario at the end where error info is freed for a caught exception, it will try to free the language exception and fail to.  We can in theory maybe implement our own IUnknown and try to handle this, but for now I won't try to handle the cycle as it is just an exception object in certain cases.